tasks/install_shell.sh

```diff
@@ -350,17 +383,31 @@ do_curl() {
 # do_fetch URL FILENAME
 do_fetch() {
   info "Trying fetch..."
-  fetch -o "$2" "$1" 2>$tmp_stderr
-  # check for bad return status
-  test $? -ne 0 && return 1
+  run_cmd "fetch -o '$2' '$1' 2>$tmp_stderr"
+  rc=$?
+
+  # check for 404
+  grep "404 Not Found" $tmp_stderr 2>&1 >/dev/null
+  if test $? -eq 0; then
+    critical "ERROR 404"
+    unable_to_retrieve_package
+  fi
```

Comment:
Curious why did we add a 404 check here? Previous code didn't seem to have it

Reply:
Confirmation that it is a valid error:
```
> eval "perl -e 'use LWP::Simple; getprint(\$ARGV[0]);' 'http://nightlies.puppet.com/bad_package_name' > 'output_file' 2>errors"
> cat errors
404 Not Found <URL:http://nightlies.puppet.com/bad_package_name>
```
I didn't research why/if it was chosen this way but since it seems like a valid error and the task would fail anyway with a more generic error, then I think it would be nice to have them all aligned and have this explicit case covered here also.